addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1478,6 +1509,193 @@ export class ListRenderer extends Component {
         }
     }
 
+   /**
+    * Create a list of virtual fields for the different property definitions of the 'properties' field
+    *
+    * @param {object} baseFieldObject The 'properties' field
+    * @param {array} propertyDefinitions relevant data to generate property 'fields'
+    * @returns {object} a list of field for the property values
+    */
+    getVirtualPropertyFields(baseFieldObject, propertyDefinitions) {
+        const fields = [];
+        for (const property of propertyDefinitions) {
+            const propertyName = this.getPropertyFieldName(property.name, baseFieldObject.name);
+            fields.push({
+                ...baseFieldObject,
+                FieldComponent: PropertyValue,
+                rawAttrs: {
+                    ...baseFieldObject.rawAttrs,
+                    name: propertyName,
+                },
+                name: propertyName,
+                label: property.string,
+                propertyType: property.type,
+                string: property.string,
+                propertyParentId: property.parentId, //used to group related columns
+                propertyParentName: property.parentName,
+            });
+        }
+        return fields;
```

Comment:
array.map / array.filter is always better than looping imo

```suggestion
        return propertyDefinitions.map(property => {
            const propertyName = this.getPropertyFieldName(property.name, baseFieldObject.name);
            return {
                ...baseFieldObject,
                FieldComponent: PropertyValue,
                rawAttrs: {
                    ...baseFieldObject.rawAttrs,
                    name: propertyName,
                },
                name: propertyName,
                label: property.string,
                propertyType: property.type,
                string: property.string,
                propertyParentId: property.parentId, //used to group related columns
                propertyParentName: property.parentName,
            };
        });
```